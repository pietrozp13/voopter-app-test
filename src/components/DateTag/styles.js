import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  dayTagsContainer: {
    flexDirection: 'row',
    margin: 2,
    borderRadius: 16,
    height: 32,
    width: 72,
    justifyContent: 'center',
    alignItems: 'center',
  },
  dayTagsText: {
    fontSize: 16,
    fontWeight: 600,
    color: '#fff',
  },
});
