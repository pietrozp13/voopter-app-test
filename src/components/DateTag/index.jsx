import { View, Text, Pressable } from 'react-native';

import { styles } from './styles';

const DAYS_COLORS = ['#8936b3', '#36b336', '#ffb84d', '#26587d'];

const BASICO_COLORS = {
  DEFAULT_BLUE: '#00aad6',
};

export default function DateTag({ date, color, onClose }) {
  console.log(date);
  const getDay = date.slice(5, 7);
  const getMonth = date.slice(8, 10);
  return (
    <View style={{ ...styles.dayTagsContainer, backgroundColor: color }}>
      <Text style={styles.dayTagsText}>
        {getDay}/{getMonth}
      </Text>
      <Pressable onPress={() => onClose(date)}>
        <Text style={styles.dayTagsText}> X</Text>
      </Pressable>
    </View>
  );
}
