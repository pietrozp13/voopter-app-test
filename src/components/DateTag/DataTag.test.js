import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react-native';

import DateTag from './index';

describe('<DateTag />', () => {
  it('has 1 child', () => {
    const tree = renderer
      .create(<DateTag key={'2023-03-10'} date={'2023-03-10'} color={'#ff0000'} />)
      .toJSON();
    expect(tree.children.length).toBe(1);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<DateTag key={'2023-03-10'} date={'2023-03-10'} color={'#ff0000'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders Hello World message on the home page', async () => {
    render(<DateTag key={'2023-03-10'} date={'2023-03-10'} color={'#ff0000'} />);
    expect(screen.getByText('10/03')).toBeDefined();
  });
});

// NOTE: You can use either:
// renderer.create(<App />) from  "react-test-renderer"
// or render(<App />) from "@testing-library/react-native"
