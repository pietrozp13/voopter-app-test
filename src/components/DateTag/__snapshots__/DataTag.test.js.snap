// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ff0000",
      "borderRadius": 16,
      "flexDirection": "row",
      "height": 28,
      "justifyContent": "center",
      "margin": 2,
      "width": 66,
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 12,
        "fontWeight": 700,
        "marginRight": 2,
      }
    }
  >
    10
    /
    03
  </Text>
</View>
`;
